pkg/apis/core/validation/validation.go

```diff
@@ -5166,45 +5156,12 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		return allErrs
 	}
 
-	if qos.GetPodQOS(oldPod) != qos.ComputePodQOS(newPod) {
```

Comment:
relaxing this validation unconditionally on pod spec update is surprising to me... I thought this PR was _removing_ the ability to change resourcey stuff via the normal pod spec update endpoint

also, xref https://github.com/kubernetes/kubernetes/pull/127744 which is attempting to *tighten* QOS validation on pod status update

Reply:
I believe we agreed to keep this check and disallow the change of QoS. @tallclair to keep me honest on the latest in all those discussions we had.

Reply:
+1 to what Anish said. This check is redundant with the check that resources are immutable, for regular pod-update. For the resize subresources, the check is still there.